src/fparser/two/tests/utils/test_bracket_base.py

```diff
@@ -0,0 +1,165 @@
+# met:
+
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''File containing unit tests for the BracketBase baseclass in
+utils.py'''
+
+import pytest
+from fparser.two.utils import BracketBase, InternalError
+from fparser.two.Fortran2003 import Name
+
+
+def test_brackets():
+    '''Test the bracketsbase match method with no content within the
+    brackets.
+
+    '''
+    for lhs, rhs in [["(", ")"], [" ( ", " ) "], ["[", "]"], ["A", "A"],
+                     ["([", "])"], ["{{[(", ")]}}"],
+                     ["  {  {  [  (  ", " ) ] } } "]]:
+        brackets = lhs + rhs
+        input_text = brackets.replace(' ', '')
+        result = BracketBase.match(brackets, None, input_text,
+                                   require_cls=False)
+        assert str(result) == "('{0}', None, '{1}')".format(
+            lhs.replace(' ', ''), rhs.replace(' ', ''))
+
+
+def test_input_too_short():
+    '''Test the bracketsbase match method returns None when the input
+    string is shorter than the brackets.
+
+    '''
+    brackets = "(())"
+    input_text = "()"
+    result = BracketBase.match(brackets, None, input_text,
+                               require_cls=False)
+    assert result is None
+
+
+def test_cls():
+    '''Test the bracketsbase match method with content within the
+    brackets for both require_cls is False and True.
+
+    '''
+    for require in [False, True]:
+        for lhs, rhs in [["(", ")"], [" ( ", " ) "], ["[", "]"], ["A", "A"],
+                         ["([", "])"], ["{{[(", ")]}}"]]:
+            brackets = lhs + rhs
+            input_text = lhs + "hello" + rhs
+            _ = BracketBase.match(brackets, Name, input_text,
```

Comment:
Should you be checking that the `match` method is not returning None?

Reply:
Nice spot. These tests would always pass.